devlib/target.py

```diff
@@ -279,6 +280,14 @@ def shutils(self):
             self._setup_shutils()
         return self._shutils
 
+    def is_running(self, process):
+        cmd = "pgrep {}"
```

Comment:
This needs to use `self.busybox`. Also just because the PID exists does not mean the task is running, it could be stopped (we may still consider that "running") or zombie (that is definitely not "running" by any metric)

Reply:
Any ideas for how to do this differently then? The intention is just to have a way to check if a background deamon is present in the system. From the kernel side the distinction between running, sleeping etc matters but this is a userspace tool so we're probably just interested in knowing whether the process exists or not.

Reply:
`busybox ps -A -o stat,pid` + `busybox awk` . You will also need to have an `as_root=False` parameter that you forward to the `target.execute()`. It's unfortunate but nothing is ever simple with POSIX process model, it's full of traps that need to be handled carefully if you don't want the API to join the cohort of half-broken stuff like `target.kick_off()`.

Otherwise you can us `target.read_file(f'/proc/{pid}/state')` and parse the output:
```
>>> cat /proc/1/status | grep State 
State:  S (sleeping)
```
if the file does not exist, then you can return false, otherwise you can return true as long as it's not a zombie